opensrp-reveal/src/main/java/org/smartregister/reveal/util/GeoJsonUtils.java

```diff
@@ -126,67 +82,139 @@ public static String getGeoJsonFromStructuresAndTasks(List<Location> structures,
                 taskProperties.put(LOCATION_UUID, structure.getProperties().getUid());
                 taskProperties.put(LOCATION_VERSION, structure.getProperties().getVersion() + "");
                 taskProperties.put(LOCATION_TYPE, structure.getProperties().getType());
-                taskProperties.put(STRUCTURE_NAME, structure.getProperties().getName());
+                interventionList.append(task.getCode());
+                interventionList.append("~");
 
             }
 
-            // The assumption is that a register structure task always exists if the structure has
-            // atleast one bednet distribution or blood screening task
-            if ( Utils.isResidentialStructure(taskProperties.get(TASK_CODE)) ) {
+            populateBusinessStatus(taskProperties, mdaStatusMap, state);
 
-                boolean familyRegTaskMissingOrFamilyRegComplete = familyRegistered || !familyRegTaskExists;
+            taskProperties.put(TASK_CODE_LIST, interventionList.toString());
+            if (structureNames.get(structure.getId()) != null) {
+                taskProperties.put(STRUCTURE_NAME, structureNames.get(structure.getId()).getStructureName());
+                taskProperties.put(FAMILY_MEMBER_NAMES, structureNames.get(structure.getId()).getFamilyMembersNames());
+            }
+            structure.getProperties().setCustomProperties(taskProperties);
 
-                if (Utils.isFocusInvestigation()) {
+        }
+        return gson.toJson(structures);
+    }
 
-                    if (familyRegTaskMissingOrFamilyRegComplete &&
-                            bednetDistributed && bloodScreeningDone) {
-                        taskProperties.put(TASK_BUSINESS_STATUS, COMPLETE);
-                    }  else if (familyRegTaskMissingOrFamilyRegComplete &&
-                             !bednetDistributed && (!bloodScreeningDone || (!bloodScreeningExists && !caseConfirmed)) ) {
-                        taskProperties.put(TASK_BUSINESS_STATUS, FAMILY_REGISTERED);
-                    } else if (bednetDistributed && familyRegTaskMissingOrFamilyRegComplete) {
-                        taskProperties.put(TASK_BUSINESS_STATUS, BEDNET_DISTRIBUTED);
-                    } else if (bloodScreeningDone) {
-                        taskProperties.put(TASK_BUSINESS_STATUS, BLOOD_SCREENING_COMPLETE);
-                    } else if (ineligibleForFamReg) {
-                        taskProperties.put(TASK_BUSINESS_STATUS, NOT_ELIGIBLE);
-                    } else {
-                        taskProperties.put(TASK_BUSINESS_STATUS, NOT_VISITED);
+    private static void calculateState(Task task, StateWrapper state, @NonNull Map<String, Integer> mdaStatusMap) {
+        if (Utils.isResidentialStructure(task.getCode())) {
+            switch (task.getCode()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A switch statement does not contain a break](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4684568)